video/out/vulkan/context_display.c

```diff
@@ -278,14 +285,84 @@ struct priv {
     struct vulkan_display_opts *opts;
     uint32_t width;
     uint32_t height;
+
+#if HAVE_DRM
+    struct mpv_opengl_drm_params_v2 drm_params;
+#endif
 };
 
+#if HAVE_DRM
+static bool open_render_fd(struct ra_ctx *ctx, const char *render_path)
+{
+    struct priv *p = ctx->priv;
+    p->drm_params.fd = -1;
+    p->drm_params.render_fd = -1;
+
+    int fd = open(render_path, O_RDWR | O_CLOEXEC);
+    p->drm_params.render_fd = fd;
+    return fd != -1;
+}
+
+static bool drm_setup(struct ra_ctx *ctx, int display_idx,
+                      VkPhysicalDevicePCIBusInfoPropertiesEXT *pci_props)
+{
+    const char *render_path = NULL;
+
+    drmDevice *devs[32] = {};
+    int count = drmGetDevices2(0, devs, MP_ARRAY_SIZE(devs));
+    for (int i = 0; i < count; i++) {
+        drmDevice *dev = devs[i];
+
+        MP_INFO(ctx, "Device %d: bustype: %d has_render: 0x%x\n",
+                i, dev->bustype, dev->available_nodes);
```

Comment:
available_nodes is a bitmask not a bool -> "has_render 0x%x" is wrong

Reply:
This was a debug line I didn't intend to include in the diff. Removed.